### PR TITLE
luci-app-statistics: add dhcpleases

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Add the dhcpleases plugin:
https://github.com/openwrt/packages/pull/14204

### CODE REVIEW

**Comment by PolynomialDivision** [comment]
Why is always this `false` added?
```
LoadPlugin dhcpleases
<Plugin dhcpleases>
	Path "/tmp/dhcp.leases"
	 false
</Plugin>
```

File: applications/luci-app-statistics/root/etc/config/luci_statistics [hunk 1]
```diff
@@ -191,3 +191,6 @@ config statistics 'collectd_thermal'
 config statistics 'collectd_uptime'
 	option enable '0'
 
+config statistics 'collectd_dhcpleases'
```
Comment (hnyman) [inline applications/luci-app-statistics/root/etc/config/luci_statistics:194]: The config section should be added to the correct alphabetical place among the input plugins

Comment (PolynomialDivision) [reply applications/luci-app-statistics/root/etc/config/luci_statistics]: changed :)

**Comment by hnyman** [comment]
I compiled this, and only then noticed that this does NOT actually provide the LuCI statistics chart. :-(
Currently it only configures the underlying collectd, right?

Looks half-complete, then.

**Comment by PolynomialDivision** [comment]
>  Currently it only configures the underlying collectd, right?

Yep, exactly. xD I currently send that to a network tool that makes graphics with it. ;)
If this is required, I will add the graphics part. ^^

**Comment by hnyman** [comment]
And that omission of the actual graph definitions seems to lead into this when one tried to config the statistics:

image


Does this config section actually work for you in LuCI?
(Or have you only tested this as editing /etc/config/... manually ?)
(Or have I miscompiled/applied it...)

**Comment by PolynomialDivision** [comment]
Whups. Sry. I just use this in our Freifunk Network that is a combination with the network stats plugin and it works without any problems. I will investigate the luci directly, when I have time! Thanks for having a look! :) I did not know that I'm crashing other stuff with this. :/

**Comment by hnyman** [comment]
Well, I did the first trial (for master):

Graph:
image

config:
image

code:
```
diff --git a/applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/dhcpleases.js b/applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/dhcpleases.js
new file mode 100644
index 000000000..cbca4a80e
--- /dev/null
+++ b/applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/dhcpleases.js
@@ -0,0 +1,24 @@
+/* Licensed to the public under the Apache License 2.0. */
+
+'use strict';
+'require baseclass';
+
+return baseclass.extend({
+	title: _('DHCP Leases'),
+
+	rrdargs: function(graph, host, plugin, plugin_instance, dtype) {
+		return {
+			title: "%H: DHCP leases",
+			vlabel: "Leases given",
+			number_format: "%5.0lf",
+			data: {
+				types: [ "count" ],
+				options: {
+					count: {
+						title: '%di'
+					}
+				}
+			}
+		};
+	}
+});
diff --git a/applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/dhcpleases.js b/applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/dhcpleases.js
new file mode 100644
index 000000000..df185f7d8
--- /dev/null
+++ b/applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/dhcpleases.js
@@ -0,0 +1,21 @@
+'use strict';
+'require baseclass';
+'require form';
+
+return baseclass.extend({
+	title: _('DHCP Leases Plugin Configuration'),
+	description: _('The dhcpleases plugin collects information about assigned DHCP leases.'),
+
+	addFormOptions: function(s) {
+		var o;
+
+		o = s.option(form.Flag, 'enable', _('Enable this plugin'));
+
+		o = s.option(form.Value, 'Path', _('DHCP leases file'));
+		o.default = '/tmp/dhcp.leases';
+	},
+
+	configSummary: function(section) {
+		return _('Monitoring DHCP leases enabled');
+	}
+});
```

**Comment by PolynomialDivision** [comment]
@hnyman That is awesome! I updated the PR and added you by
```
Added-Frontend-by: Hannu Nyman >hannu.nymanohnoyoudont@iki.fi>
```
is that okay?

**Comment by hnyman** [comment]
Never seen that before:  hannu.nymanohnoyoudont@iki.fi

Some copy-paste protection you apply?

**Comment by PolynomialDivision** [comment]
> Never seen that before: [hannu.nymanohnoyoudont@iki.fi](mailto:hannu.nymanohnoyoudont@iki.fi)

Whups, sry. Do u want to be mentioned somehow? Or do u want to do the PR?

I will now compile everything and test it.

**Comment by hnyman** [comment]
you can mention me, but just correct the email ;-)

I left one more comment about sorting to the packages repo, but otherwise this seems to work, so once you fix that, we could merge them in.

**Comment by PolynomialDivision** [comment]
> I left one more comment about sorting to the packages repo, but otherwise this seems to work, so once you fix that, we could merge them in.

Changed everything. ;) Let me test again and then we can merge. :D

And then the same for ipv6 statistics... xD

**Comment by PolynomialDivision** [comment]
For me it works. Tested right now with latest trunk on wdr4300.

**Comment by hnyman** [comment]
Applied as f8552afb

**Pull request closed without merging**